### PR TITLE
Blog App -validations and model specs.

### PR DESCRIPTION
**Pull request by devendra-alt**
## 👨‍💻  [Blog App -validations and model specs.](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md)

- add `validation` for table attributes in models
- add `rspec` to project
- performed `spec` tests for given models 
- tested `validation` and `methods`
- added `after_save` callback on update methods

### CODE REVIEW

File: spec/model/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  it 'should return post_count as zero' do
+    expect(user.posts.size).to be(0)
+  end
+  it 'should return post_count as one' do
+    post = Post.create(title: 'hello', text: 'hello world test', author_id: user.id)
+    expect(post.user.posts_count).to be(1)
+    expect(post.user.posts.size).to be(1)
+  end
```
Comment (cynthiainga) [inline spec/model/user_spec.rb:23]: - [x] Good job so far but you should test the scenario where if the posts_count is not an integer to not be valid 🙃

File: spec/model/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through
+    the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from
+    sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and Evil) by Cicero,
+    written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance.
+    The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", comes from a line in section 1.10.32.'
+    expect(post).to_not be_valid
+  end
+  it 'should have post counter to 1' do
+    expect(post.user.posts_count).to be(1)
+  end
```
Comment (cynthiainga) [inline spec/model/post_spec.rb:30]: - [x] As mentioned in the requirements `CommentsCounter` and `LikesCounter` must be tested
  - must be an integer greater than or equal to zero, and don't forget to test the scenario if the `comments_counter` and `likes_counter` are not an integer 😁

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @devendra-alt, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [x] No linters errors ✔️ 
- [x] All implemented tests are passing ✔️ 
- [x] Correct Gitflow ✔️ 

## Required Changes ♻️

- [x] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:
- I think it would be nice if you added the command to run the project locally and how to run tests locally in your README file 🙃 🙃 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @devendra-alt 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Changes requested have been implemented ✅ 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**